qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -375,6 +375,15 @@ def solve(self, problem: QuadraticProgram) -> ADMMOptimizationResult:
                                         variables=base_result.variables,
                                         state=self._state)
 
+        # convert back integer to binary
+        result = cast(ADMMOptimizationResult, int2bin.decode(result))
+
+        # check for feasibility
+        if QuadraticProgram.is_feasible(problem, result.x):
+            result.status = OptimizationResultStatus.SUCCESS
```

Comment:
You don't have to call `QuadraticProgram.is_feasible(problem, result.x)`. This looks like you call a static method. A shorter version `problem.is_feasible(result.x)` should be better.

Also `status = OptimizationResultStatus.SUCCESS if problem.is_feasible(result.x) else OptimizationResultStatus.INFEASIBLE` looks shorter?

Reply:
> 
> 
> You don't have to call `QuadraticProgram.is_feasible(problem, result.x)`. This looks like you call a static method. A shorter version `problem.is_feasible(result.x)` should be better.
> 
Sure, that looks great. 

> Also `status = OptimizationResultStatus.SUCCESS if problem.is_feasible(result.x) else OptimizationResultStatus.INFEASIBLE` looks shorter?

I implemented the above once but I didn't wanted it to be confusing so I implemented the normal if...else...  . I will revert back to this implementation.